pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -1091,10 +1087,6 @@ def split_on_constraints(self, source_id: NodeIdLike, constraints: Iterable[KInn
         source = self.node(source_id)
         branch_node_ids = [self.create_node(source.cterm.add_constraint(c)).id for c in constraints]
         csubsts = [not_none(source.cterm.match_with_constraint(self.node(id).cterm)) for id in branch_node_ids]
-        csubsts = [
-            reduce(CSubst.add_constraint, flatten_label('#And', constraint), csubst)
-            for (csubst, constraint) in zip(csubsts, constraints, strict=True)
-        ]
```

Comment:
No need to flatten, since it's already flattened during `CSubst` construction.

Reply:
Additionally, we don't need to add the alpha constraints manually, because of the new `match_with_constraint`.